src/main/java/com/alipay/remoting/ConnectionEventListener.java

```diff
@@ -54,12 +57,16 @@ public void onEvent(ConnectionEventType type, String remoteAddress, Connection c
      */
     public void addConnectionEventProcessor(ConnectionEventType type,
                                             ConnectionEventProcessor processor) {
-        List<ConnectionEventProcessor> processorList = this.processors.get(type);
-        if (processorList == null) {
-            this.processors.putIfAbsent(type, new ArrayList<ConnectionEventProcessor>(1));
-            processorList = this.processors.get(type);
+        lock.lock();
```

Comment:
As mentioned in #309, it is ArrayList not thread safe. I think it is better to replace ArrayList to a thread safe List implementation such as CopyOnWriteArrayList instead of using a lock.

Reply:
use Collections.synchronizedList() will be ok and please add a UT

Reply:
@zjcscut will you adjust it based on the above suggestions?